lib/pcp/client.rb

```diff
@@ -53,6 +53,11 @@ def connect(seconds = 0)
         raise "Cannot be run on the same thread as the reactor"
       end
 
+      if @connection
+        # We close over so much, we really just need to say no for now
+        raise "Can only connect once per client"
```

Comment:
Ruby exceptions have call stacks in them, so we should at least see that the raise came from pcp/client.rb:58.  I think that's enough.